pyproject.toml

```diff
@@ -26,7 +26,7 @@ numpy = [
 sqlalchemy = { version = ">=2.0.21", optional = true }
 openpyxl = "^3.0.10"
 alembic = { version = "^1.0.11", optional = true }
-urllib3 = ">=1.0"
+urllib3 = ">=1.26"
```

Comment:
My only concern with this pin version is anyone using an older urllib3 than 1.26, as these were the users who voiced issues when we originally released with the version pinned to `urllib3>=2.0.0`.

Looking through past issues, it seems that as long as we still support `urllib3==1.27.0` then the previous issues would still be avoided:

- [evidence 1](https://github.com/databricks/databricks-sql-python/issues/203)
- [evidence 2](https://github.com/databricks/databricks-sql-python/issues/190)

I also checked the default version of `urllib3` in a DBSQL notebook and it's 1.26.14, so I think this change is acceptable.